### DISCUSSION TITLE
EFORMS-NOTICE-EDITOR

### DISCUSSION THREAD

**Discussion by rengit62** [OP-TED/eForms-SDK#324] [Q&A] [answered]
Hi Labs !
I downloaded Eforms-notice-editor, very nice for showing how making a new notice.
Can you give a suggestion about editing an existing notice ?

Thanks a lot, and regards
Renato

**Comment by rouschr** [comment, marked as answer]
There is an ongoing feature for this but it is incomplete, so you will need some patience :)
We will demonstrate loading of notice XML in the next releases (after SDK 1.6.0 comes out).
There are also plans to demonstrate various types of validation, like showing error messages.
As you can imagine there is a link between the form fields and the fields.json xpath, the form sections and the node "ND-".
The challenges are the repeatability, loading order and various edge cases.

**Reply by rengit62** [reply]
Thanks Christophe !
Anyway I made  a valid xml notice subtype 1 ( valid against XSDs ) and I sent it to CVS validation, 
getting back status 200 and an **huge** message with tag type <svrl:failed-assert  .. .
I solved all except 4 that I believe depend on the data input in the form.
I would like to know this  : 
how can I run locally these tests , for example, prompting some errors in the browser after the form submission ?
As for now there are no example in the Demo , are there examples in other projects ? 
I send a zip with notice xml and the srvl response file

[Notice1.zip](https://github.com/OP-TED/eForms-SDK/files/10759272/Notice1.zip)

Thanks again and Regards
Renato

**Reply by rouschr** [reply]
> prompting some errors in the browser after the form submission

We have plans to work on something like this soon as validation is a central topic.
Getting back svrl in the editor demo was the first step, we intend to parse the svrl XML, build a custom JSON and respond with the JSON. The web page will read that JSON and attach errors to the concerned fields ...

**Reply by rengit62** [reply]
sorry Christophe, I was not clear to explain .
I meant ...for example a message like :  <svrl:text>Notice Dispatch Date (BT-05) value must be earlier than Notice Publication Date Preferred (BT-738) value.</svrl:text>
and
 "assert" : {
      "value" : "{ND-Root} ${TRUE}",
      "severity" : "ERROR",
      "constraints" : [ {
        "value" : "{ND-Root} ${(BT-05(a)-notice < BT-738-notice) or not(BT-738-notice is present)}",
        "severity" : "ERROR",
        "message" : "rule|text|BR-BT-00005-0150"
      }, {
It is possible exec "{ND-Root} ${(BT-05(a)-notice < BT-738-notice) or not(BT-738-notice is present)}" **BEFORE** create ConceptualModel, PhysicalModel sending to CVS Validation ?
I would like to implement something like : 
**Notice validation inside your app**
I guess that is an EFX expression, there are sample to exec that rule, or  there will samples in 1.6 version ?
Thanks
Renato

**Reply by rouschr** [reply]
What you want is CVS to validate your notice.
Your application sends the XML to CVS, CVS responds.
How you treat the response and attach the report to form fields is something we will demo later.

> It is possible exec "{ND-Root} ${(BT-05(a)-notice < BT-738-notice) or not(BT-738-notice is present)}" BEFORE create ConceptualModel

you could in theory execute EFX in the browser using javascript, maybe we will also demo this in the future but we are not there yet.
Another possibility is to execute EFX on the back-end / server-side using the provided Java EFX implementation.
Some rules are easier to execute dynamically (in the browser) and other rules are better suited for execution on the server.
It is possible that at some point we will categorize rules to help with that.
